### ISSUE TITLE
Spaces get condensed by --fix when they are alone in a <span>

### ISSUE THREAD

**Issue by pbondoer** [yannickcr/eslint-plugin-react#1717]
We often need to add spacing between two elements in JSX. For example:

```jsx
<GenericTextComponent type='test' />
<span>{' '}</span>
<GenericTextComponent type='other' />
```

Applying `--fix` turns the code into this:
```jsx
<GenericTextComponent type='test' />
<span />
<GenericTextComponent type='other' />
```
which is not the same thing.

I think I've traced down what's happening. The code above gives off this error:
```
error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
```

Which, when corrected (by removing the curly braces and quotes), gives off another error:
```
error  Empty components are self-closing  react/self-closing-comp
```

which `--fix` happily turns into the `<span />` above.

Versions:
* `eslint@4.18.2`
* `eslint-plugin-react@7.7.0`

**Comment by ljharb** [comment]
Ouch, that’s a problem. The bug is with curly brace presence (they are necessary around whitespace) and maybe also with self-closing-comp (inline elements with a space inside aren’t necessarily empty).

These need to be fixed soon.

**Comment by ljharb** [comment]
cc @sharmilajesupaul

**Comment by sharmilajesupaul** [comment]
It looks like `react/jsx-curly-brace-presence` is erroneously warning on `{' '}` which is necessary even if `children`/global option is set to `never`. Once that's removed, `react/self-closing-comp` is autofixing the `<span>` to be self-closing.

The  `react/jsx-curly-brace-presence` rule doesn't have a test for this case but  `react/self-closing-comp` [does have a test case that covers whitespace.](https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/self-closing-comp.js#L44) 

So fixing `react/jsx-curly-brace-presence` this will solve this issue... I think 🤔

**Comment by ljharb** [comment]
We still should fix react-self-closing-comp to not treat inline HTML elements as empty when they contain whitespace (it should autofix them to a single space, perhaps)
